### PR TITLE
Swagger spec file for the resource api

### PR DESCRIPTION
**Pull request by Matt-Guest**
Includes endpoints and data models.

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+info:
+  description: "An API used to query information about resources"
+  version: "1.0.0"
+  title: "ONS Resource API"
+  license:
+    name: "Open Government Licence v3.0"
+    url: "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+basePath: "/v1"
+tags:
+- name: "Resources"
```
Comment (CarlHembrough) [inline swagger.yaml]: Resources is very a general and overloaded term. It may be worth having a discussion as a group to determine if there is something more specific we can call it.

File: swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    in: path
+    type: string
+    required: true
+  codeId:
+    name: codeId
+    description: "CodeId that represents a type within a resource"
+    in: path
+    type: string
+    required: true
+paths:
```
Comment (CarlHembrough) [inline swagger.yaml]: consider a /resource endpoint for listing all resources.

File: swagger.yaml (hunk 1 above)
Comment (Matt-Guest) [inline swagger.yaml]: After a group discussion we renamed the API to  data dictionary.

File: swagger.yaml (hunk 2 above)
Comment (Matt-Guest) [inline swagger.yaml]: Done

File: swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    required: true
+  query:
+    name: query
+    description: "Query all data dictionaries"
+    in: body
+    required: true
+    schema:
+      $ref: '#/definitions/Query'
+paths:
+  /datadictionary:
```
Comment (eldeal) [inline swagger.yaml]: Possibly hyphenate: `/data-dictionary`?

Comment (Matt-Guest) [inline swagger.yaml]: Added a hyphen and renamed the API.

File: swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    url: "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+basePath: "/v1"
+tags:
+- name: "Code List"
+schemes:
+- "http"
+parameters:
+  id:
+    name: id
+    description: "Id that represents a list code"
```
Comment (eldeal) [inline swagger.yaml]: Is this an ID for list of codes, or a value within the list?

File: swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    in: path
+    type: string
+    required: true
+  codeId:
+    name: codeId
+    description: "CodeId that represents a type within a code list"
```
Comment (eldeal) [inline swagger.yaml]: By a type do we mean a discrete value? A single code within the list?

File: swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+            $ref: '#/definitions/CodeList'
+        400:
+          description: "Missing parameters within request"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /code-list/search:
+    post:
+      tags:
+      - "Code List"
+      summary: "Get a set of code lists containing information about a dimensions"
```
Comment (eldeal) [inline swagger.yaml]: A dimensions? No `a` or no `s` please :)

File: swagger.yaml (hunk 4 above)
Comment (Matt-Guest) [inline swagger.yaml]: This should be an ID for a code list. I reword the it.

File: swagger.yaml (hunk 5 above)
Comment (Matt-Guest) [inline swagger.yaml]: Yes thats right. I reword this to make it clearer.

File: swagger.yaml (hunk 6 above)
Comment (Matt-Guest) [inline swagger.yaml]: Fixed the typo :(
